--- a/src/x11/window.rs
+++ b/src/x11/window.rs
@@ -10,6 +10,7 @@
     HasRawDisplayHandle, HasRawWindowHandle, RawDisplayHandle, RawWindowHandle, XlibDisplayHandle,
     XlibWindowHandle,
 };
+extern crate rwh04;
 
 use x11rb::connection::Connection;
 use x11rb::protocol::xproto::{
@@ -120,15 +121,17 @@
 impl<'a> Window<'a> {
     pub fn open_parented<P, H, B>(parent: &P, options: WindowOpenOptions, build: B) -> WindowHandle
     where
-        P: HasRawWindowHandle,
+        // Accept the rwh 0.4 trait so that iced_baseview (which pins 0.4) can
+        // call this function.
+        P: rwh04::HasRawWindowHandle,
         H: WindowHandler + 'static,
         B: FnOnce(&mut crate::Window) -> H,
         B: Send + 'static,
     {
         // Convert parent into something that X understands
         let parent_id = match parent.raw_window_handle() {
-            RawWindowHandle::Xlib(h) => h.window as u32,
-            RawWindowHandle::Xcb(h) => h.window,
+            rwh04::RawWindowHandle::Xlib(h) => h.window as u32,
+            rwh04::RawWindowHandle::Xcb(h) => h.window,
             h => panic!("unsupported parent handle type {:?}", h),
         };
 
@@ -166,6 +169,30 @@
         });
     }
 
+    /// Create a window without a parent. Like `open_parented`, this returns
+    /// immediately without blocking the calling thread; unlike `open_parented`,
+    /// the window is created as a child of the X root window (there is no
+    /// "view without a window" concept on X11).
+    pub fn open_as_if_parented<H, B>(options: WindowOpenOptions, build: B) -> WindowHandle
+    where
+        H: WindowHandler + 'static,
+        B: FnOnce(&mut crate::Window) -> H,
+        B: Send + 'static,
+    {
+        let (tx, rx) = mpsc::sync_channel::<WindowOpenResult>(1);
+
+        let (_, mut window_handle) = ParentHandle::new();
+
+        thread::spawn(move || {
+            Self::window_thread(None, options, build, tx, None).unwrap();
+        });
+
+        let raw_window_handle = rx.recv().unwrap().unwrap();
+        window_handle.raw_window_handle = Some(raw_window_handle.0);
+
+        window_handle
+    }
+
     fn window_thread<H, B>(
         parent: Option<u32>, options: WindowOpenOptions, build: B,
         tx: mpsc::SyncSender<WindowOpenResult>, parent_handle: Option<ParentHandle>,
