diff --git a/src/wrappers/appkit/view.rs b/src/wrappers/appkit/view.rs
index 3c04777..26c8b49 100644
--- a/src/wrappers/appkit/view.rs
+++ b/src/wrappers/appkit/view.rs
@@ -3,7 +3,7 @@ use objc2::runtime::AnyObject;
 use objc2::{msg_send, Encoding, Message, RefEncode};
 use objc2_app_kit::{NSDragOperation, NSDraggingInfo, NSEvent, NSView, NSWindow};
 use objc2_core_foundation::CGRect;
-use objc2_foundation::{NSNotification, NSPoint};
+use objc2_foundation::{NSNotification, NSPoint, NSSize};
 use raw_window_handle::{AppKitWindowHandle, HasRawWindowHandle, RawWindowHandle};
 use std::ffi::{c_void, CStr};
 use std::marker::PhantomData;
@@ -118,6 +118,7 @@ pub trait ViewImpl: Sized {
     fn hit_test(this: ViewRef<'_, Self>, point: NSPoint) -> Option<&NSView>;
     fn view_will_move_to_window(this: ViewRef<Self>, new_window: Option<&NSWindow>);
     fn update_tracking_areas(this: ViewRef<Self>);
+    fn set_frame_size(this: ViewRef<Self>, new_size: NSSize);
     fn mouse_moved(this: ViewRef<Self>, event: &NSEvent);
     fn scroll_wheel(this: ViewRef<Self>, event: &NSEvent);
     fn dragging_entered(
